numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1527,8 +1527,16 @@ OBJECT_to_@TOTYPE@(void *input, void *output, npy_intp n,
  * #oskip = 1*18,(PyArray_DESCR(aop)->elsize)*3,1*2,
  *          1*18,(PyArray_DESCR(aop)->elsize)*3,1*2,
  *          1*18,(PyArray_DESCR(aop)->elsize)*3,1*2#
+ * #convert = 1*14, 0, 1*3, 0*3, 1*2,
+ *            1*14, 0, 1*3, 0*3, 1*2,
```

Comment:
@charris, this reverts the commit (except the tests). With this only `longdouble` is affected by the fixup, for which it fixes a bug, and is not just a bit of a jumping through hoops.

The two lines above should be the only change remaining from the original backport., i.e. the original fix effectively set `convert = 0` for *all* types. This one only does it for `longdouble`.